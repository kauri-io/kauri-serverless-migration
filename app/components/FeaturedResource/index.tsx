import styled from 'styled-components'
import UserAvatarComponent, {
    IProps as UserAvatarComponentProps,
} from '../UserAvatar'
import { Label, Title2, BodyCard } from '../Typography'
import TagList from '../Tags/TagList'
import SecondaryButtonComponent from '../Button/SecondaryButton'
import slugify from 'slugify'

const ContentSection = styled.div`
    display: flex;
    @media (max-width: ${props => props.theme.breakpoints[0]}) {
        flex-direction: column;
        > :first-child {
            margin-bottom: ${props => props.theme.space[3]}px;
        }
    }
    ${props => props.theme.padContent};
`

const ResourceDetailsContainer = styled.section`
    display: flex;
    flex-direction: column;
    flex: 3;
    > :not(:last-child) {
        margin-bottom: ${props => props.theme.space[2]}px;
    }
`

const ViewContainer = styled.section`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    flex: 1;
`

export const FeaturedResourceContainer = styled.div`
    display: flex;
    width: 100%;
    background: white;
    flex-direction: column;
    padding: ${props => props.theme.space[3]}px 0px;
    background: white;
    @media (max-width: ${props => props.theme.breakpoints[0]}) {
        padding: ${props => props.theme.space[3]}px
            ${props => props.theme.space[2]}px;
        ${ViewContainer} {
            align-items: flex-start;
        }
    }
`

interface IProps {
    id: string
    version: string | null
    title: string
    description: string
    tags: string[]
    resourceType: string
    ownerResourceType: string
    linkComponent: (
        children: React.ReactElement<any>,
        route: string
    ) => React.ReactElement<any>
}

const FeaturedResource: React.FunctionComponent<
    IProps & UserAvatarComponentProps
> = ({
    title,
    description,
    linkComponent,
    resourceType,
    id,
    userId,
    tags,
    username,
    avatar,
    ownerResourceType,
}) => {
    const slug = slugify(title, { lower: true })
    return (
        <FeaturedResourceContainer>
            <ContentSection>
                <ResourceDetailsContainer>
                    <Label>Featured</Label>
                    {linkComponent(
                        <>
                            <Title2>{title && String(title)}</Title2>
                            <BodyCard>
                                {description && String(description)}
                            </BodyCard>
                        </>,
                        resourceType === 'article'
                            ? `/${slug}/${id}/a`
                            : resourceType === 'collection'
                            ? `/${slugify(name, { lower: true })}/${id}/c`
                            : `/community/${id}`
                    )}
                    <TagList maxTags={3} color="textPrimary" tags={tags} />
                    {linkComponent(
                        <UserAvatarComponent
                            userId={userId}
                            username={username}
                            avatar={avatar}
                        />,
                        ownerResourceType === 'COMMUNITY'
                            ? `/community/${userId}`
                            : `/public-profile/${userId}`
                    )}
                </ResourceDetailsContainer>
                <ViewContainer>
                    {linkComponent(
                        <SecondaryButtonComponent
                            border="primary"
                            color="textPrimary"
                        >{`View ${resourceType}`}</SecondaryButtonComponent>,
                        resourceType === 'article'
                            ? `/${slug}/${id}/a`
                            : resourceType === 'collection'
                            ? `/${slugify(name, { lower: true })}/${id}/c`
                            : `/community/${id}`
                    )}
                </ViewContainer>
            </ContentSection>
        </FeaturedResourceContainer>
    )
}

export default FeaturedResource
