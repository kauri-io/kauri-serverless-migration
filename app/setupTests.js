import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import theme from './lib/theme-config'
import { ThemeProvider } from 'styled-components'

Enzyme.configure({ adapter: new Adapter() })

const observeMock = {
    observe: () => null,
    disconnect: () => null, // maybe not needed
}

window.IntersectionObserver = () => observeMock

export const mountWithTheme = children =>
    mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const renderWithTheme = children =>
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const shallowWithTheme = children =>
    shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
