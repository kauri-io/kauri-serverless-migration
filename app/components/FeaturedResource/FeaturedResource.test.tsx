import FeaturedResource from './index'
import { mountWithTheme } from '../../setupTests'

let props = {
    id: '11',
    version: null,
    title: 'Test Title',
    description: 'test description',
    tags: ['test'],
    resourceType: 'ARTICLE',
    ownerResourceType: 'COMMUNITY',
    linkComponent: jest.fn(),
    avatar: 'avatar url',
}

describe('components/FeaturedResource', () => {
    it('snapshot should match', () => {
        const wrapper = mountWithTheme(<FeaturedResource {...props} />)
        expect(wrapper).toMatchSnapshot()
    })
})
