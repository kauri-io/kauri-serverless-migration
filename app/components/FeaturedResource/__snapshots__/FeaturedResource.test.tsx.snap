// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/FeaturedResource snapshot should match 1`] = `
<ThemeProvider
  theme={
    Object {
      "DEFAULT_CARD_WIDTH": 288,
      "MAIN_BODY_WIDTH": 1242,
      "bg": Object {
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "hoverTextColor": "#108E72",
        "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      },
      "bgPrimary": "#1E2428",
      "bgSecondary": "#1E3D3B",
      "borderTextArea": "#979797",
      "breakpoints": Array [
        "500px",
        "52em",
        "1242px",
      ],
      "colors": Object {
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "hoverTextColor": "#108E72",
        "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      },
      "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
      "disabledBackgroundColor": "#D6D6D6",
      "disabledTextColor": "#9B9B9B",
      "divider": "#EBEBEB",
      "errorRedColor": "#C03030",
      "fontSizes": Array [
        11,
        13,
        14,
        16,
        18,
        20,
        22,
        28,
        32,
      ],
      "fontWeight": Array [
        300,
        "normal",
        500,
        700,
      ],
      "hoverTextColor": "#108E72",
      "padContent": "
    max-width: 1242px;
    margin-left: auto;
    margin-right: auto;
    @media (min-width: 1242px) {
      min-width: 1242px;
    }
    @media (max-width: 1242px) {
      min-width: 90%;
    }
  ",
      "padding": "calc((100vw - 1242px) / 2)",
      "paddingTop": "2.5em",
      "primary": "#0BA986",
      "primaryColor": "#0BA986",
      "primaryDark": "#11856B",
      "primaryTextColor": "#1E2428",
      "secondary": "#11856B",
      "secondaryBlue": "rgb(81, 123, 162, 1)",
      "secondaryBlueDark": "rgb(81, 118, 142, 1)",
      "secondaryColor": "#1E3D3B",
      "secondaryTextColor": "#283035",
      "space": Array [
        0,
        10,
        15,
        30,
        60,
      ],
      "tertiaryBackgroundColor": "#F7F7F7",
      "textPrimary": "#1E2428",
      "white": "#FFFFFF",
    }
  }
>
  <FeaturedResource
    avatar="avatar url"
    description="test description"
    id="11"
    linkComponent={
      [MockFunction] {
        "calls": Array [
          Array [
            <React.Fragment>
              <ForwardRef(Typography__styledComponent)>
                Test Title
              </ForwardRef(Typography__styledComponent)>
              <ForwardRef(Typography__component)>
                test description
              </ForwardRef(Typography__component)>
            </React.Fragment>,
            "/community/11",
          ],
          Array [
            <UserAvatarComponent
              avatar="avatar url"
            />,
            "/community/undefined",
          ],
          Array [
            <SecondaryButtonComponent
              border="primary"
              color="textPrimary"
            >
              View ARTICLE
            </SecondaryButtonComponent>,
            "/community/11",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    ownerResourceType="COMMUNITY"
    resourceType="ARTICLE"
    tags={
      Array [
        "test",
      ]
    }
    title="Test Title"
    version={null}
  >
    <FeaturedResource__FeaturedResourceContainer>
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "FeaturedResource__FeaturedResourceContainer-sc-4xfnje-3",
              "isStatic": false,
              "lastClassName": "hGKkgc",
              "rules": Array [
                "display:flex;width:100%;background:white;flex-direction:column;padding:",
                [Function],
                "px 0px;background:white;@media (max-width:",
                [Function],
                "){padding:",
                [Function],
                "px ",
                [Function],
                "px;",
                ".FeaturedResource__ViewContainer-sc-4xfnje-2",
                "{align-items:flex-start;}}",
              ],
            },
            "displayName": "FeaturedResource__FeaturedResourceContainer",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "FeaturedResource__FeaturedResourceContainer-sc-4xfnje-3",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="FeaturedResource__FeaturedResourceContainer-sc-4xfnje-3 hGKkgc"
        >
          <FeaturedResource__ContentSection>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "FeaturedResource__ContentSection-sc-4xfnje-0",
                    "isStatic": false,
                    "lastClassName": "gCNcvx",
                    "rules": Array [
                      "display:flex;@media (max-width:",
                      [Function],
                      "){flex-direction:column;>:first-child{margin-bottom:",
                      [Function],
                      "px;}}",
                      [Function],
                      ";",
                    ],
                  },
                  "displayName": "FeaturedResource__ContentSection",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "FeaturedResource__ContentSection-sc-4xfnje-0",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="FeaturedResource__ContentSection-sc-4xfnje-0 gCNcvx"
              >
                <FeaturedResource__ResourceDetailsContainer>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "FeaturedResource__ResourceDetailsContainer-sc-4xfnje-1",
                          "isStatic": false,
                          "lastClassName": "ffAtLy",
                          "rules": Array [
                            "display:flex;flex-direction:column;flex:3;>:not(:last-child){margin-bottom:",
                            [Function],
                            "px;}",
                          ],
                        },
                        "displayName": "FeaturedResource__ResourceDetailsContainer",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "FeaturedResource__ResourceDetailsContainer-sc-4xfnje-1",
                        "target": "section",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <section
                      className="FeaturedResource__ResourceDetailsContainer-sc-4xfnje-1 ffAtLy"
                    >
                      <Typography__component>
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "Typography__component-sc-36xkf6-0",
                                "isStatic": false,
                                "lastClassName": "kALGmD",
                                "rules": Array [
                                  "font-size:11px;font-weight:bold;text-transform:uppercase;color:",
                                  [Function],
                                  ";text-align:",
                                  [Function],
                                  ";",
                                  [Function],
                                  " ",
                                  [Function],
                                ],
                              },
                              "displayName": "Typography__component",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "Typography__component-sc-36xkf6-0",
                              "target": "span",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                        >
                          <span
                            className="Typography__component-sc-36xkf6-0 kALGmD"
                          >
                            Featured
                          </span>
                        </StyledComponent>
                      </Typography__component>
                      <TagList
                        color="textPrimary"
                        maxTags={3}
                        tags={
                          Array [
                            "test",
                          ]
                        }
                      >
                        <TagList__Container>
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "TagList__Container-ipfhd9-0",
                                  "isStatic": false,
                                  "lastClassName": "ihYPdv",
                                  "rules": Array [
                                    "display:flex;margin:",
                                    "10",
                                    "px 0 0 0;flex-direction:",
                                    [Function],
                                    ";flex-wrap:wrap;justify-content:",
                                    [Function],
                                    ";",
                                  ],
                                },
                                "displayName": "TagList__Container",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "TagList__Container-ipfhd9-0",
                                "target": "div",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <div
                              className="TagList__Container-ipfhd9-0 ihYPdv"
                            >
                              <TagList__StyledTag
                                color="textPrimary"
                                hiddenTags={false}
                                key="0"
                                onClick={[Function]}
                              >
                                <StyledComponent
                                  color="textPrimary"
                                  forwardedComponent={
                                    Object {
                                      "$$typeof": Symbol(react.forward_ref),
                                      "attrs": Array [],
                                      "componentStyle": ComponentStyle {
                                        "componentId": "TagList__StyledTag-ipfhd9-1",
                                        "isStatic": false,
                                        "lastClassName": "eGuhSG",
                                        "rules": Array [
                                          "cursor:pointer;transition:0.3s;&:not(:last-child):after{content:'•';color:",
                                          "#0BA986",
                                          ";margin:",
                                          "5",
                                          "px;font-weight:",
                                          "700",
                                          ";}&:hover{color:",
                                          "#0BA986",
                                          ";}",
                                          [Function],
                                          ";",
                                          [Function],
                                          ";",
                                        ],
                                      },
                                      "displayName": "TagList__StyledTag",
                                      "foldedComponentIds": Array [],
                                      "render": [Function],
                                      "styledComponentId": "TagList__StyledTag-ipfhd9-1",
                                      "target": "div",
                                      "toString": [Function],
                                      "warnTooManyClasses": [Function],
                                      "withComponent": [Function],
                                    }
                                  }
                                  forwardedRef={null}
                                  hiddenTags={false}
                                  onClick={[Function]}
                                >
                                  <div
                                    className="TagList__StyledTag-ipfhd9-1 eGuhSG"
                                    color="textPrimary"
                                    onClick={[Function]}
                                  >
                                    test
                                  </div>
                                </StyledComponent>
                              </TagList__StyledTag>
                            </div>
                          </StyledComponent>
                        </TagList__Container>
                      </TagList>
                    </section>
                  </StyledComponent>
                </FeaturedResource__ResourceDetailsContainer>
                <FeaturedResource__ViewContainer>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "FeaturedResource__ViewContainer-sc-4xfnje-2",
                          "isStatic": true,
                          "lastClassName": "bQDGjh",
                          "rules": Array [
                            "display:flex;flex-direction:column;justify-content:center;align-items:center;flex:1;",
                          ],
                        },
                        "displayName": "FeaturedResource__ViewContainer",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "FeaturedResource__ViewContainer-sc-4xfnje-2",
                        "target": "section",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <section
                      className="FeaturedResource__ViewContainer-sc-4xfnje-2 bQDGjh"
                    />
                  </StyledComponent>
                </FeaturedResource__ViewContainer>
              </div>
            </StyledComponent>
          </FeaturedResource__ContentSection>
        </div>
      </StyledComponent>
    </FeaturedResource__FeaturedResourceContainer>
  </FeaturedResource>
</ThemeProvider>
`;
